### ISSUE TITLE
Attacking Rooks (probably wrong graph creation)

### ISSUE THREAD

**Issue by ChrisVilches** [ChrisVilches/Algorithms#4]
When connecting a row with a column, the first nodes are rows, and the next ones are columns, but I connect rows with rows (it still works - accepted).

Why? Should something be refactored?

**Comment by ChrisVilches** [comment]
From the point of view of a strictly bipartite graph (i.e. it's always bipartite, without the possibility of it not being so), it makes sense though. Every node from the set is connected to nodes from the other set, but without the need of explicitly specifying they are from the other set, because they always are (no matter what).

**Comment by ChrisVilches** [comment]
In this example I coded it the same way:

https://github.com/ChrisVilches/Algorithms/blob/main/kattis/escapeplan.cpp

In this other program, both sets contain the same N elements, so it cannot be programmed that way (but in this case it's easy - no need to add or count which element is from which side since it doesn't matter):

https://github.com/ChrisVilches/Algorithms/blob/main/kattis/bookclub.cpp

**Comment by ChrisVilches** [comment]
Note: Still need to refactor Attacking Rooks to remove extra graph size.

**Comment by ChrisVilches** [comment]
However, I'm not sure if the nodes need to be separated into two sets explicitly, since the bipartite matching algorithm may need to differentiate the sets clearly. (Need to study the algorithm better)

**Comment by ChrisVilches** [comment]
Blood groups also:
https://github.com/ChrisVilches/Algorithms/blob/main/urionlinejudge/2014-blood_groups.cpp

Same graph creation without adding an offset for the next set of nodes.

**Comment by ChrisVilches** [comment]
Conclusion:

The matrix is always using Edmonds Matrix notation (in my case, list of adjacency lists). It may look like a normal graph, but the edges are always pointing to nodes that may not necessarily exist in the range of `[0, g.size() - 1]`.

In other words, we could have `vector<vector<int>> g(3)` but add the edge `g[0].push_back(1000)`. This would work.

The reason is that the matching algorithm never starts a `bpm` traversal from the node `1000`. It'd always use nodes `0, 1, 2` (in the graph above with size `3`). The recursive `bpm` uses `match[v]`, but this value is a node of the left set (the one with size 3).

So it's safe to implement it with node offset, or without node offset. It works perfectly OK in both cases.

NOTE: I didn't study all my codes, nor I tried to refactor all BPM programs. I only refactored some, but I only refactored the BPM algorithm rather than the graph creation, which was actually OK.
